src/multidecoder/decoders/network.py

```diff
@@ -1107,7 +1172,95 @@ def freq_of_most_common(data: bytes) -> float:
     )
 
 
+def extract_html_attribute_value(data: bytes, start_index: int) -> tuple[bytes | None, int, int]:
+    if start_index >= len(data):
+        return None, start_index, 0
+
+    accepted_quotes = b"\"'`"
+    block_url_starts = string.punctuation.encode()
+
+    delim = bytes([data[start_index]])
+
+    if delim in accepted_quotes:
+        # A valid container / delimiter character is used.
+        m = re.match(rb"(\r|\n|[^\p{C}" + delim + rb"])*" + delim, data, re.S, start_index + 1)
+
+        if m:
+            return data[start_index + 1: m.end() - 1], m.end(), 1
+    elif delim.isascii() and delim not in block_url_starts:
+        # The text is not contained in quotes. The end is implicitly defined by parsing pattern.
+        contents = HTML_UNQUOTED_ATTR_VALUE_PATTERN.search(data, start_index)
```

Comment:
I think this should be match instead of search. It isn't clear that the pattern can't match outside the attribute